web-client/integration-tests/adcClerkViewsMessagesJourney.test.js

```diff
@@ -0,0 +1,296 @@
+      await cerebralTest.runSequence('gotoMessagesSequence', {
+        box: 'inbox',
+        queue: messageQueue,
+      });
+
+      const messages = cerebralTest.getState('messages');
+
+      await markMessageAsComplete(
+        cerebralTest,
+        messages,
+        message6Subject,
+        message6SubjectFromState,
+      );
+
+      await markMessageAsComplete(
+        cerebralTest,
+        messages,
+        message7Subject,
+        message7SubjectFromState,
+      );
+
+      await refreshElasticsearchIndex();
+    });
+
+    it(`verify default sorting of ${messageQueue} completed box completedAt sort field, descending`, async () => {
+      const { completedMessages } = await getUserMessageCount(
+        cerebralTest,
+        'completed',
+        messageQueue,
+      );
+
+      let afterCompletedMessageCount = cerebralTest.getState('messages').length;
+
+      const expected = [message7SubjectFromState, message6SubjectFromState];
+
+      expect(afterCompletedMessageCount).toEqual(
+        expected.length + beforeCompletedMessageCount,
+      );
+
+      validateMessageOrdering(completedMessages, expected);
+    });
+
+    loginAs(cerebralTest, 'docketclerk@example.com');
+    it('verify the table headers are not clickable for non-ADC users', async () => {
+      const showSortableHeaders = withAppContextDecorator(
+        showSortableHeadersComputed,
+      );
+
+      await cerebralTest.runSequence('gotoMessagesSequence', {
+        box: 'inbox',
+        queue: messageQueue,
+      });
+
+      const isTableHeaderClickable = runCompute(showSortableHeaders, {
+        state: cerebralTest.getState(),
+      });
+
+      expect(isTableHeaderClickable).toBe(false);
+    });
+    it('verify the messages in sent box are sorted from newest to oldest non-ADC users', async () => {
+      await cerebralTest.runSequence('gotoMessagesSequence', {
+        box: 'outbox',
+        queue: messageQueue,
+      });
+
+      const { messages } = runCompute(formattedMessagesComputed, {
+        state: cerebralTest.getState(),
+      });
+
+      const newSortedMessages = [...messages].sort((a, b) =>
+        b.createdAt.localeCompare(a.createdAt),
+      );
+
+      expect(messages).toEqual(newSortedMessages);
+    });
+  });
+});
+
+const validateMessageOrdering = (actualMessages, expectedMessageSubjects) => {
+  // Iterating over the inboxMessages array verifies that we
+  // found the expected messages in the order we expected them to be.
+  // The expectation on pointer verifies the count of expected messages.
+  let pointer = 0;
+  actualMessages.forEach(message => {
+    if (message.subject === expectedMessageSubjects[pointer]) {
+      pointer++;
+    }
+  });
+  expect(pointer).toEqual(expectedMessageSubjects.length);
+};
+
+const markMessageAsComplete = async (
+  context,
+  messages,
+  messageCompletedSubject,
+  completedMessageSubject,
+) => {
+  const foundMessage = messages.find(
```

Comment:
❓ Is there a way to save the `messageId`s from each of the messages created during test setup? That way we can be sure we are grabbing the message we intend off of state. I see that the date is included in the subject, which is a good way of uniquely identifying the messages! Just wondering if there is a more direct way.

Reply:
Yes. The work has been refactored to use ids.